docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,189 @@ weight: 700
+| `attributes` map of the OAuth2 ID token.                                                                                                                                | Set `email_attribute_name` configuration option. By default, Grafana searches for email under `email:primary` key. |
+| `upn` field of the OAuth2 ID token.                                                                                                                                     | N/A                                                                                                                |
+| `email` field of the user information from the UserInfo endpoint.                                                                                                       | N/A                                                                                                                |
+| Another field of the user information from the UserInfo endpoint.                                                                                                       | Set `email_attribute_path` configuration option.                                                                   |
+| Email address marked as primary from the `/emails` endpoint of <br /> the OAuth2 provider (obtained by appending `/emails` to the URL <br /> configured with `api_url`) | N/A                                                                                                                |
+
+## Configure a refresh token
 
 > **Note:** This feature is behind the `accessTokenExpirationCheck` feature toggle.
 
-When a user logs in using an OAuth provider, Grafana verifies that the access token has not expired. When an access token expires, Grafana uses the provided refresh token (if any exists) to obtain a new access token.
+When a user logs in using an OAuth2 provider, Grafana verifies that the access token has not expired. When an access token expires, Grafana uses the provided refresh token (if any exists) to obtain a new access token.
 
 Grafana uses a refresh token to obtain a new access token without requiring the user to log in again. If a refresh token doesn't exist, Grafana logs the user out of the system after the access token has expired.
 
-To configure Generic OAuth to use a refresh token, perform one or both of the following tasks, if required:
+To configure generic OAuth2 to use a refresh token, perform one or both of the following steps, if required:
 
-- Extend the `[auth.generic_oauth]` section with additional scopes
-- Enable the refresh token on the provider
+1. Extend the `scopes` field of `[auth.generic_oauth]` section in Grafana configuration file with additional scopes.
+1. Enable the refresh token on the provider.
 
-### Configure automatic login
+## Configure role mapping
```

Comment:
@IevaVasiljeva , should this be an H3 so that it is nested under "Configuration options"?

Reply:
Role mapping is a step that has caused a lot of pain to customers, and that customers sometimes configure separately from the initial auth provider integration setup. So I think it's good to have it more visible.

Reply:
👍